test/parallel/test-source-map-cjs-require-cache.js

```diff
@@ -0,0 +1,32 @@
+// Flags: --enable-source-maps --expose-gc
+
+/**
+ * This test verifies that the source map of a CJS module is cleared after the
+ * CJS module is reclaimed by GC.
+ */
+
+'use strict';
+const common = require('../common');
+const assert = require('node:assert');
+const { findSourceMap } = require('node:module');
+
+const moduleId = require.resolve('../fixtures/source-map/no-throw.js');
+
+function run() {
+  require(moduleId);
+  delete require.cache[moduleId];
+  const idx = module.children.findIndex((child) => child.id === moduleId);
+  assert.ok(idx >= 0);
+  module.children.splice(idx, 1);
+
+  // Verify that the source map is still available
+  assert.notStrictEqual(findSourceMap(moduleId), undefined);
+}
+
+// Run the test in a function scope so that every variable can be reclaimed by GC.
+run();
+
+// Run until the source map is cleared by GC, or fail the test after determined iterations.
+common.gcUntil('SourceMap of deleted CJS module is cleared', () => {
```

Comment:
I think this has the potential of flakes because the strategy used by `common.gcUntil()` isn't very sound (i.e. sometimes `gc()` is not aggressive enough to purge things). It would be safer to do it for a bunch more modules (say 30), and checks if any of them can be collected, which reduces the likelihood of the one single module that we are looking at just cannot be garbage collected in time because V8 just doesn't feel pressured enough to do that.

Reply:
Thanks for the suggestion, updated the test with more GC pressure to reduce the chance of flaky.

Reply:
Maybe we should launch a stress-test CI to make sure we're not introducing any flakiness